python/cuml/cuml_accel_tests/test_estimator_proxy.py

```diff
@@ -559,3 +560,72 @@ def test_method_that_only_exists_on_cpu_estimator():
     assert hasattr(model._cpu, "n_features_in_")
     # Method was run on host
     assert scipy.sparse.issparse(model.coef_)
+
+
+@pytest.mark.parametrize(
+    "methods",
+    [
+        "set_output,fit,transform",
+        "fit,set_output,transform",
+        "set_output,fit_transform",
+    ],
+)
+def test_set_output(methods):
```

Comment:
The tests here check the fiddliest bits of `set_output` support in detail. Some of these details require estimator-specifics, but are also generic to how `ProxyBase` is implemented. Testing these for only one estimator seems sufficient to me.